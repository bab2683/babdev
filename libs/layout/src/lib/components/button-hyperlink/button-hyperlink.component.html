<a [href]="value" target="_blank">
  <fa-icon [icon]="icon"></fa-icon>
</a>
