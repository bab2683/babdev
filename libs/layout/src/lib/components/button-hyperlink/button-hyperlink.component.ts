import { Component } from '@angular/core';
import { IconDefinition } from '@fortawesome/fontawesome-svg-core';
import { faLink } from '@fortawesome/free-solid-svg-icons';

@Component({
  selector: 'babdev-button-hyperlink',
  templateUrl: './button-hyperlink.component.html',
  styleUrls: ['./button-hyperlink.component.scss']
})
export class ButtonHyperlinkComponent {
  public icon: IconDefinition = faLink;
  public value: string;
}
