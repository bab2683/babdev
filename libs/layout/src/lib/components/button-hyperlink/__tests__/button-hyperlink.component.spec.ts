import { NO_ERRORS_SCHEMA } from '@angular/core';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { ButtonHyperlinkComponent } from '../button-hyperlink.component';

describe('ButtonHyperlinkComponent', () => {
  let component: ButtonHyperlinkComponent;
  let fixture: ComponentFixture<ButtonHyperlinkComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ButtonHyperlinkComponent],
      schemas: [NO_ERRORS_SCHEMA]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ButtonHyperlinkComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
