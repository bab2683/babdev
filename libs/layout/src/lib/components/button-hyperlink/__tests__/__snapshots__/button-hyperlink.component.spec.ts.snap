// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonHyperlinkComponent should match snapshot 1`] = `
<babdev-button-hyperlink
  icon={[Function Object]}
>
  <a
    href="null"
    target="_blank"
  >
    <fa-icon />
  </a>
</babdev-button-hyperlink>
`;
